ring-core/src/ring/util/response.clj

```diff
@@ -44,6 +44,14 @@
       :headers {"Location" url}
       :body    body}))
 
+(defn bad-request
+  "Returns a Ring response for a HTTP 400 created response."
```

Comment:
"400 created response" should probably be "400 'bad request' response".

Ideally make it consistant with `not-found`:

    "Returns a 400 'bad request' response."